go/vt/vtgate/planbuilder/testdata/filter_cases.txt

```diff
@@ -3927,3 +3927,114 @@ Gen4 plan same as above
   }
 }
 Gen4 plan same as above
+
+# optimize ORs to IN route op codes #1
+"select col from user where id = 1 or id = 2"
+{
+  "QueryType": "SELECT",
+  "Original": "select col from user where id = 1 or id = 2",
+  "Instructions": {
+    "OperatorType": "Route",
+    "Variant": "Scatter",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "FieldQuery": "select col from `user` where 1 != 1",
+    "Query": "select col from `user` where id = 1 or id = 2",
+    "Table": "`user`"
+  }
+}
+{
+  "QueryType": "SELECT",
+  "Original": "select col from user where id = 1 or id = 2",
+  "Instructions": {
+    "OperatorType": "Route",
+    "Variant": "IN",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "FieldQuery": "select col from `user` where 1 != 1",
+    "Query": "select col from `user` where id = 1 or id = 2",
+    "Table": "`user`",
+    "Values": [
+      "(INT64(1), INT64(2))"
+    ],
+    "Vindex": "user_index"
+  }
+}
+
+# optimize ORs to IN route op codes #2
+"select col from user where id = 1 or id = 2 or id = 3"
+{
+  "QueryType": "SELECT",
+  "Original": "select col from user where id = 1 or id = 2 or id = 3",
+  "Instructions": {
+    "OperatorType": "Route",
+    "Variant": "Scatter",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "FieldQuery": "select col from `user` where 1 != 1",
+    "Query": "select col from `user` where id = 1 or id = 2 or id = 3",
+    "Table": "`user`"
+  }
+}
+{
+  "QueryType": "SELECT",
+  "Original": "select col from user where id = 1 or id = 2 or id = 3",
+  "Instructions": {
+    "OperatorType": "Route",
+    "Variant": "IN",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "FieldQuery": "select col from `user` where 1 != 1",
+    "Query": "select col from `user` where id = 1 or id = 2 or id = 3",
+    "Table": "`user`",
+    "Values": [
+      "(INT64(1), INT64(2), INT64(3))"
+    ],
+    "Vindex": "user_index"
+  }
+}
+
+# optimize ORs to IN route op codes #3
+"select col from user where (id = 1 or id = 2) or (id = 3 or id = 4)"
+{
+  "QueryType": "SELECT",
+  "Original": "select col from user where (id = 1 or id = 2) or (id = 3 or id = 4)",
+  "Instructions": {
+    "OperatorType": "Route",
+    "Variant": "Scatter",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "FieldQuery": "select col from `user` where 1 != 1",
+    "Query": "select col from `user` where id = 1 or id = 2 or (id = 3 or id = 4)",
+    "Table": "`user`"
+  }
+}
+{
+  "QueryType": "SELECT",
+  "Original": "select col from user where (id = 1 or id = 2) or (id = 3 or id = 4)",
+  "Instructions": {
+    "OperatorType": "Route",
+    "Variant": "IN",
+    "Keyspace": {
+      "Name": "user",
+      "Sharded": true
+    },
+    "FieldQuery": "select col from `user` where 1 != 1",
+    "Query": "select col from `user` where id = 1 or id = 2 or (id = 3 or id = 4)",
```

Comment:
I don't understand why the query is not rewritten with the `IN` comparison. Are we printing the original query, but execute the query `IN` instead of `=`s ?

Reply:
we're not changing the query, only the routing